### ISSUE TITLE
Very dangerous command in hack/local_up_karmada.sh

### ISSUE THREAD

**Issue by LronDC** [karmada-io/karmada#3707]
**What happened**:
I'm installing Karmada via ***hack/local_up_karmada.sh*** on an environment that already has other clusters installed with kind and has workloads working, something unexpected happened. The script deleted all clusters.

```shell
(base) ❱❱❱ bash hack/local-up-karmada.sh
Preparing: 'kind' existence check - passed
Preparing: 'kubectl' existence check - passed
Preparing kindClusterConfig in path: /tmp/tmp.GqkzyfZP2r
Deleted all clusters and the log file is in /tmp/karmada/delete-all-clusters.log
Creating cluster karmada-host and the log file is in /tmp/karmada/karmada-host.log
```
**What you expected to happen**:
Do not delete clusters not related to Karmada.

**How to reproduce it (as minimally and precisely as possible)**:
- kind create one
- bash hack/local_up_karmada.sh

and one gone.

**Anything else we need to know?**:
Actually, it's in ***util.sh*** line 444:
```shell
function util::delete_all_clusters() {
  local main_config=${1}
  local member_config=${2}
  local log_path=${3}

  local log_file="${log_path}"/delete-all-clusters.log
  rm -rf ${log_file}
  mkdir -p ${log_path}

  kind delete clusters --all >> "${log_file}" 2>&1
  rm -f "${main_config}"
  rm -f "${member_config}"
```
`delete clusters --all` that's too rough.
**Environment**:
- Karmada version:
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
- Others:

**Comment by LronDC** [comment]
I think this kind of operation should be prompted and confirmed, or use conditions to filter out karmada-related clusters, and then operate

**Comment by RainbowMango** [comment]
Yeah, It would be great if `hack/local_up_karmada.sh` could only delete the kind cluster created by itself.

> I think this kind of operation should be prompted and confirmed, or use conditions to filter out karmada-related clusters, and then operate.

I tend to the latter one. 

> Expecting the community to make a decision and I'll work on it.

Just go for it. Appreciate.
